plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/shipping/index.tsx

```diff
@@ -1,165 +1,52 @@
-						( prefersCollection &&
-							isPackageRateCollectable( rate ) &&
-							rate.selected ) ||
-						// Or the shopper does not prefer collection and the rate is selected
-						( ! prefersCollection && rate.selected )
-				)
-				.flatMap( ( rate ) => rate.name );
-		}
-	);
-	const addressComplete = isAddressComplete( shippingAddress, [
-		'state',
-		'country',
-		'postcode',
-		'city',
-	] );
-	const shippingMethodsMissing = areShippingMethodsMissing(
-		hasRates,
-		prefersCollection,
-		shippingRates
-	);
-
-	const valueToDisplay =
-		totalShippingValue === 0 ? (
-			<strong>{ __( 'Free', 'woocommerce' ) }</strong>
-		) : (
-			totalShippingValue
-		);
 
+export const TotalsShipping = ( {
+	label = __( 'Shipping', 'woocommerce' ),
+	placeholder = null,
+	collaterals = null,
+}: TotalShippingProps ): JSX.Element | null => {
+	const { cartTotals, shippingRates } = useStoreCart();
+	const hasRates = hasShippingRate( shippingRates );
 	return (
-		<div
-			className={ clsx(
-				'wc-block-components-totals-shipping',
-				className
-			) }
-		>
-			<ShippingCalculatorContext.Provider
-				value={ {
-					showCalculator,
-					shippingCalculatorID: 'shipping-calculator-form-wrapper',
-					isShippingCalculatorOpen,
-					setIsShippingCalculatorOpen,
-				} }
-			>
-				<TotalsItem
-					label={ __( 'Delivery', 'woocommerce' ) }
-					value={
-						! shippingMethodsMissing && cartHasCalculatedShipping
-							? // if address is not complete, display the link to add an address.
-							  valueToDisplay
-							: ( ! addressComplete || isCheckout ) && (
-									<ShippingPlaceholder
-										showCalculator={ showCalculator }
-										isCheckout={ isCheckout }
-										addressProvided={ addressComplete }
-									/>
-							  )
-					}
-					description={
-						( ! shippingMethodsMissing &&
-							cartHasCalculatedShipping ) ||
-						// If address is complete, display the shipping address.
-						( addressComplete && ! isCheckout ) ? (
-							<>
-								<ShippingVia
-									selectedShippingRates={
-										selectedShippingRates
-									}
-								/>
-								<ShippingAddress
-									shippingAddress={ shippingAddress }
-								/>
-							</>
-						) : null
-					}
-					currency={ currency }
-				/>
-				{ showShippingCalculatorForm && <ShippingCalculator /> }
-				{ showRateSelector &&
-					cartHasCalculatedShipping &&
-					! showShippingCalculatorForm && (
-						<ShippingRateSelector
-							hasRates={ hasRates }
-							shippingRates={ shippingRates }
-							isLoadingRates={ isLoadingRates }
-							isAddressComplete={ addressComplete }
-							shippingAddress={ shippingAddress }
-						/>
-					) }
-			</ShippingCalculatorContext.Provider>
+		<div className="wc-block-components-totals-shipping">
+			<TotalsItem
+				label={ label }
+				value={
+					hasRates
+						? renderShippingTotalValue( cartTotals )
+						: placeholder
+				}
+				description={
+					<>
```

Comment:
I find it awkward that we tend to favor passing in JSX trees versus just using children in many situations. Since the pattern has already been perpetuated there is not much that would be easy to refactor here, but I'd really prefer if we build components to take children instead.

At least here we either:

a. Don't need Shipping Collaterals at all, it's just a wrapper to add a classname, and used exactly once?
b. If you prefer to keep the component, just use `children` to render children instead of perpetuating pattern of passing children as props which is non-idiomatic.

Reply:
I've removed the component in https://github.com/woocommerce/woocommerce/pull/52313/commits/a2996fdee798a535f3d9290bf16182c59e9a719e

`TotalsItem` is used in other places do I won't touch that in this PR.